packages/snap/src/keyring.ts

```diff
@@ -320,32 +231,39 @@ export class SimpleKeyring implements Keyring {
 
   #handleSigningRequest(method: string, params: Json): Json {
     switch (method) {
-      case 'personal_sign': {
-        const [from, message] = params as string[];
+      case EthMethod.PersonalSign: {
+        const [message, from] = params as [string, string];
         return this.#signPersonalMessage(from, message);
       }
 
-      case 'eth_sendTransaction':
-      case 'eth_signTransaction':
-      case SigningMethods.SignTransaction: {
-        const [from, tx, opts] = params as [string, JsonTx, Json];
-        return this.#signTransaction(from, tx, opts);
+      case EthMethod.SignTransaction: {
+        const [tx] = params as [any];
+        return this.#signTransaction(tx);
+      }
+
+      case EthMethod.SignTypedData:
+      case EthMethod.SignTypedDataV1: {
+        const [from, data] = params as [string, Json];
+        return this.#signTypedData(from, data, {
+          version: SignTypedDataVersion.V1,
+        });
       }
 
-      case 'eth_signTypedData':
-      case 'eth_signTypedData_v1':
-      case 'eth_signTypedData_v2':
-      case 'eth_signTypedData_v3':
-      case 'eth_signTypedData_v4': {
-        const [from, data, opts] = params as [
-          string,
-          Json,
-          { version: SignTypedDataVersion },
-        ];
-        return this.#signTypedData(from, data, opts);
```

Comment:
Is there a reason to make this `switch-case` logic more verbose?

Reply:
Only to decode the version of the typed data (1, 3 or 4). I removed the `opts` parameter since it's not part of the standard, so we have to use the method name to get the right version number.